vertical-pod-autoscaler/recommender/signals/prometheus_client.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package signals
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+)
+
+// A client that talks to Prometheus using its HTTP API.
+type PrometheusClient interface {
```

Comment:
Why are we not using GRPC v2 api?

Reply:
It's not listed on the official docs: https://prometheus.io/docs/prometheus/latest/querying/api/. I also couldn't find much information about it, other than the implementation on github, so I'm not sure what state it's in (presumably under development).